### ISSUE TITLE
nested dictionary from nested group in HDF5

### ISSUE THREAD

**Issue by ssomnath** [pycroscopy/sidpy#99]
I think it would be good to have a function that can read nested groups + attributes as a nested dictionary

**Comment by gduscher** [comment]
I agree, if we abandon flattening the dictionary  this would make pyNSID more compatible with  NeXus,
and the hdf5 files would be easier to read.

**Comment by gduscher** [comment]
Should be adressed in pyNSID

**Comment by ssomnath** [comment]
Actually, this is a generic enough function that it should probably stay in sidpy. I think we should keep clearly NSID specific stuff in pyNSID.
